executor/slow_query.go

```diff
@@ -663,7 +663,20 @@ func (e *slowQueryRetriever) parseLog(ctx context.Context, sctx sessionctx.Conte
 func (e *slowQueryRetriever) setColumnValue(sctx sessionctx.Context, row []types.Datum, tz *time.Location, field, value string, checker *slowLogChecker, lineNum int) bool {
 	factory := e.columnValueFactoryMap[field]
 	if factory == nil {
-		return true
+		switch {
+		// Fix issue 34320, when slow log time is not in the output columns, the time filter condition is mistakenly discard.
+		case field == variable.SlowLogTimeStr && checker != nil:
+			t, err := ParseTime(value)
+			if err != nil {
+				err = fmt.Errorf("Parse slow log at line %v, failed field is %v, failed value is %v, error is %v", lineNum, field, value, err)
+				sctx.GetSessionVars().StmtCtx.AppendWarning(err)
+				return false
+			}
+			timeValue := types.NewTime(types.FromGoTime(t), mysql.TypeTimestamp, types.MaxFsp)
+			return checker.isTimeValid(timeValue)
```

Comment:
Do we need to check `checker != nil`? I see we do it in `getColumnValueFactoryByName`.
Besides, do we need to handle it with timezone？

Reply:
`checker != nil` has already been check before entering this code branch. See https://github.com/pingcap/tidb/pull/34322/files#diff-9cb39ec0c8d4960b6e8f64d27731000897c8296503bad061a574b3dd8ac17c11R668

> do we need to handle it with timezone

You can see this special order https://github.com/pingcap/tidb/blob/ca900e6a8381bc7e55717c65e4031e5961c97092/executor/slow_query.go#L715-L725
I asked @crazycs520 is this a bug, should it be handle timezone first and `check()` then... he confirmed this is not bug.

So it means `check()` use the unhandled timezone...

Reply:
Got it.